### PR TITLE
Adds "Curator" as a librarian alt-title.

### PR DESCRIPTION
**Pull request by Eneocho**
We now have a decent painting system and many have gone and expressed themselves with it. Adding a dedicated title for the art enthusiast felt right.

Curators spawn with a regular crayon box, a palette, a chisel (to make sculptures or wall engravings) and 40 wood. 40 wood is enough for 4 wooden blocks (and thus 4 statues), 10 small canvas, 6 landscape/portrait canvas or 8 large canvas. 

[content][role]
:cl:
 * rscadd: New Librarian alt title: "Curator". Curators spawn with a crayon box, a chisel, a palette, 40 wood and a beret.

### CODE REVIEW

File: code/datums/outfit/civilian.dm [hunk 1]
```diff
@@ -571,8 +571,16 @@
 	alt_title_items_to_collect = list(
 		"Game Master" = list(
 			/obj/item/weapon/storage/pill_bottle/dice/with_die = GRASP_LEFT_HAND,
+		),
+		"Artist" = list(
+			/obj/item/stack/sheet/wood/bigstack = null     //should spawn on the backpack
```
Comment (SonixApache) [inline code/datums/outfit/civilian.dm]: ,

File: code/datums/outfit/civilian.dm [hunk 2]
```diff
@@ -571,8 +571,16 @@
...
+			/obj/item/stack/sheet/wood/bigstack = null
```
Comment (SonixApache) [inline code/datums/outfit/civilian.dm]: ,

**Comment by Kurfursten** [comment]
Artist doesn't seem quite right. Curator seems to be better driving at what you're looking for, as they could use the library to display art pieces. It would be nice if you supported the role by making it possible to upload art to the library database.

**Comment by Eneocho** [comment]
>It would be nice if you supported the role by making it possible to upload art to the library database.

Lord knows I would love to do that but I've no idea how. They could upload the template string to a book I guess? That would be more of a job for @Jellyveggie than me, since it's his canvas code.

>Artist doesn't seem quite right. Curator seems to be better driving at what you're looking for, as they could use the library to display art pieces. 

I thought about curator too, but curators don't usually make their paintings, they oversee the paintings, select and sometimes restore, but not really paint themselves.

File: code/datums/outfit/civilian.dm (hunk 1 above)
Comment (Eneocho) [reply code/datums/outfit/civilian.dm]: Damn, I didn't notice. Thanks for the catch.

**Comment by jellyveggie2** [comment]
Templates are meant to be shared between players to recreate or even alter paintings, for this I'd say you should look at /datum`/custom_painting` and store the relevant data: `bitmap`, `bitmap_width`, `bitmap_height`, and likely also `author`, `title `and `description`. 
To load it, you create a `/datum/custom_painting` and load all that info back in, then have the object that holds the datum call `render_on()` to render the updated bitmap on the target icon.
The way it works right now with the canvas object (`/obj/structure/painting/custom/` and it's item `/obj/item/mounted/frame/painting/custom/`) just creating a canvas already creates the datum, and calling a canvas' `update_painting(render=TRUE)` after replacing the bitmap would call `render_on()`.

As to actually saving that data across rounds... I've no idea how SQL works in BYOND, from what I've been able to look at so far, understanding how the code in lib_machines.dm works seems to be the way, though with paintings having their own table in the database.

Mechanically, I've no idea yet how it should work. Maybe piggyback off the librarian's machines to scan/print paintings, maybe create new machines, or maybe even go the way TG did and have paintings be saved only if they're hanging on specific walls while the round ends (Dunno yet how they retrieve saved paintings afterwards, other than I believe there's a chance for saved custom paintings to spawn in place of mapped default paintings). 

Haven't been able to look into it much more in-depth yet, between painting bugfixes, failing to figure out color quantization so templates on the wiki made with crayons aren't a hot mess of a thousand functionally-identical shades of the same color, and other requests either from the orignal PR or that keep popping up. 

Also, you got the "wrong" JellyVeggie, that account's been dead for at least a couple years now.

**Comment by Eneocho** [comment]
>Also, you got the "wrong" JellyVeggie, that account's been dead for at least a couple years now.

oops

**Comment by PavelCIA** [comment]
Give them a beret

**Comment by Eneocho** [comment]
> Give them a beret

good call

**Comment by Blithering** [comment]
I'd rather it as a mime alt title, pretentious Frenchman wearing a beret could be either a mine or a painter.

**Comment by ShiftyRail** [comment]
See previous comments. This is a good idea, but a Mime alt-title would definitively fit better. Or a Curator alt title.

**Comment by Eneocho** [comment]
I thought about the mime angle, but I don't like it. First and foremost because it would take the role away from the guy that actually wants to play mime. Second, the intention was to give an incentive for joining as a librarian alt title (kinda like journalists or game masters), so the guy who just wants to paint for the entire shift doesn't eat another guy's job just to paint.
I guess I could concede on the curator, but I still think that curators are more of an administrative/managerial role rather than artist themselves.

**Comment by Eneocho** [comment]
@ShiftyRail Done.
Nervestation's having some issues on the map testing, #31770 got affected too.

**Comment by Kurfursten** [comment]
I think this is the right direction to go, and having art in the library computer for future round recovery gives the librarian some much needed love related to his actual job that he has been aching for years to receive. Just needs someone to code it.

**Comment by Eneocho** [comment]
If we ever get art in the librarian computer, should it be a template or a duplicate of the original? If template, then it ain't that useful for curating. If duplicate, nobody would use the template system.

**Comment by Kurfursten** [comment]
You could do either, but I was thinking that when artwork gets uploaded we just convert it to a flat image and upload that image. This picture would be a duplicate of the art on canvas but no longer able to be edited. So people can't just print out art as a starting point for their own picture.

**Comment by ghost** [comment]
\>\>\>/tg/

**Pull request merged**